### ISSUE TITLE
arrow-body-style suggestion: newline option

### ISSUE THREAD

**Issue by mockdeep** [eslint/eslint#8506]
* **ESLint Version:** 3.19.0

I would love some sort of a `multi-line-and-as-needed` configuration for the `arrow-body-style` rule that requires braces when there is more than one line, but not when it fits on one line:

```js
// good
collection.map((item) => item.some_property);

// good
collection.map((item) => {
  return item.some_long_property_name + something_else * whatever;
});

// bad
collection.map((item) => { return item.some_property; });

// bad
collection.map((item) =>
  item.some_long_property_name + something_else * whatever);
```

The no-braces shorthand is nice on a single line for brevity, but I prefer the closure and explicitness of braces when something spans multiple lines.

**Comment by ilyavolodin** [comment]
`arrow-body-style` is a rule to enforce/prohibit braces around arrow function body. It doesn't really care about linebreaks, or any statements other then `return` inside function's body. So this would be a wrong rule to add this option to.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.

**Comment by mbrevda** [comment]
Which rule would this be a better fit for? 

It seems the same idea already exists for blocks following an  [if/else/while/do](https://eslint.org/docs/rules/curly#multi-or-nest). Perhaps the block body of an arrow function should be given the same treatment? Or is there a better place for this rule?

**Comment by not-an-aardvark** [comment]
I think this could be achieved with a combination of [`implicit-arrow-linebreak`](https://eslint.org/docs/rules/implicit-arrow-linebreak) and [`brace-style`](https://eslint.org/docs/rules/brace-style).

**Comment by mbrevda** [comment]
Been playing with those all day - could not get it to work (notice they are both enabled in the sandbox link above). Would love to be proven otherwise!
